zbd.c

```diff
@@ -1283,8 +1295,24 @@ static void zbd_put_io(const struct io_u *io_u)
 	       "%s: terminate I/O (%lld, %llu) for zone %u\n",
 	       f->file_name, io_u->offset, io_u->buflen, zone_idx);
 
-	ret = pthread_mutex_unlock(&z->mutex);
-	assert(ret == 0);
+	if (td->o.zone_append && z->pending_ios > 0) {
+		pthread_mutex_lock(&z->mutex);
+		z->pending_ios--;
```

Comment:
pending_ios is updated under a lock but tested outside of the lock... This smells like atomicity problems for some arch... shouldn't pending_ios be an atomic (and then beware of compilation issues for other OSes/compilers).

Reply:
Sure, this code will be refactored in the v2 patches.